Frameworks/CoreText/CTFont.mm

```diff
@@ -811,8 +811,13 @@ void CTFontGetVerticalTranslationsForGlyphs(CTFontRef font, const CGGlyph glyphs
         }
 
         std::transform(metrics, metrics + count, translations, [font](const DWRITE_GLYPH_METRICS& metrics) {
-            return CGSize{ -0.5 * __CTFontScaleMetric(font, metrics.advanceWidth),
-                           -__CTFontScaleMetric(font, metrics.verticalOriginY - metrics.topSideBearing + sc_VerticalTranslationSpace) };
+            if (metrics.advanceWidth == metrics.rightSideBearing && metrics.verticalOriginY == metrics.topSideBearing) {
+                // Special case of empty glyph, should just return CGRectZero;
+                return CGSizeZero;
+            }
+
+            return CGSize{ -__CTFontScaleMetric(font, metrics.advanceWidth >> 1),
```

Comment:
mm, I understand that this >> 1 is probably safe, but the gains here don't seem worth overriding principle of least surprise.